From b685cff6eb5a0aa445484fd19f0773f8b96b5c85 Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Fri, 10 Jul 2015 09:20:11 +0800
Subject: [PATCH] fs: direct-io: convert rw from READ to KERNEL_READ

KERNEL_READ equal READ puls REQ_KERNEL, which indicates
this brq comes from direct-io. Any block-layer devices who
care this req type can get it now.

Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 fs/direct-io.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/fs/direct-io.c b/fs/direct-io.c
index 7ab90f5081ee..6e5dd6f5df1a 100644
--- a/fs/direct-io.c
+++ b/fs/direct-io.c
@@ -370,6 +370,7 @@ static inline void dio_bio_submit(struct dio *dio, struct dio_submit *sdio)
 {
 	struct bio *bio = sdio->bio;
 	unsigned long flags;
+	int rw;
 
 	bio->bi_private = dio;
 
@@ -380,6 +381,9 @@ static inline void dio_bio_submit(struct dio *dio, struct dio_submit *sdio)
 	if (dio->is_async && dio->rw == READ)
 		bio_set_pages_dirty(bio);
 
+	rw = dio->rw;
+	dio->rw |= (dio->rw == READ) ? KERNEL_READ : KERNEL_WRITE;
+
 	if (sdio->submit_io)
 		sdio->submit_io(dio->rw, bio, dio->inode,
 			       sdio->logical_offset_in_bio);
-- 
2.35.3

